forge-game/src/main/java/forge/game/ability/effects/AnimateEffectBase.java

```diff
@@ -132,6 +159,7 @@ public static void doAnimate(final Card c, final SpellAbility sa, final Integer
         final List<Trigger> addedTriggers = Lists.newArrayList();
         for (final String s : triggers) {
             final Trigger parsedTrigger = TriggerHandler.parseTrigger(AbilityUtils.getSVar(sa, s), c, false, sa);
+            if (perpetual) parsedTrigger.setCardState(c.getCurrentState());
```

Comment:
I'm not sure if this is necessary? Flames of Moradin and Goblin Trapfinder EOT phase triggers still don't work with it

Without it, cardstate at this point is Flames/Trapfinder and NOT the card being animated to.

But again, not sure it really matters